runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -9993,28 +9976,22 @@ VMinlineCompareAndSwap(TR::Node *node, TR::CodeGenerator *cg, TR::InstOpCode::Mn
          decompressedValueNode = decompressedValueNode->getFirstChild();
          }
 
-      if (decompressedValueNode->getOpCode().isSub() || TR::Compiler->vm.heapBaseAddress() == 0 || newVNode->isNull())
+      isValueCompressedReference = true;
```

Comment:
Can we fold uses of this as true?

Reply:
Don't think so. This is initialized as `false`. If the function parameter `isObj` is false, then this boolean stays false as well. Folding it as `true` can lead to unwanted behaviour when it's used on line 9981 and 9993 below